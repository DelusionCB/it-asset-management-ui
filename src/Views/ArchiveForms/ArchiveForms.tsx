import React, {useEffect, useState} from 'react';
import {useLocation} from 'react-router-dom';
import FormFields from '../../Components/FormFields/FormFields';
import './index.scss'

function ArchiveForms (): JSX.Element {
    const params = useLocation();
    const [loading, isLoading] = useState(true);
    const [mode, setMode] = useState('')

    useEffect(() => {
        isLoading(true)
        if (params.state) {
            setMode(params.state.action)
            isLoading(false)
        } else {
            setMode('create')
            isLoading(false)
        }
    }, [params]);

    return (
        <div className='formviewer'>
            <FormFields
                loading={loading}
                mode={mode}
                params={params}
            />
        </div>
    );
}

export default ArchiveForms;
