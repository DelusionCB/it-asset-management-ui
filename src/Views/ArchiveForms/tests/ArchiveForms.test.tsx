import React from 'react';
import * as TestRenderer from 'react-test-renderer';
import {BrowserRouter} from 'react-router-dom'
import '@testing-library/jest-dom'
import ArchiveForms from '../ArchiveForms';

describe('<ArchiveViewer />', () => {
    let renderer: TestRenderer.ReactTestRenderer;

    // Render the component before tests
    beforeAll(() => {
        renderer = TestRenderer.create(
            <BrowserRouter>
                <ArchiveForms />
            </BrowserRouter>,
        );
    });
    describe('Snapshot', () => {
        it('renders correctly', () => {
            expect(renderer.toJSON()).toMatchSnapshot();
        })
    })
})
