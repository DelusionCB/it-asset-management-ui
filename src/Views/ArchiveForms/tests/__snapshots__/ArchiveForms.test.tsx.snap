// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<ArchiveViewer /> Snapshot renders correctly 1`] = `
<div
  className="formviewer"
>
  <div
    animation="border"
    className="spinner-border"
    role="status"
  >
    <span
      className="visually-hidden"
    >
      Ladataan...
    </span>
  </div>
</div>
`;
